### PR TITLE
Add SASS/SCSS preprocessor

### PR DESCRIPTION
**Pull request by PRTTMPRPHT**
fyi all.

+5 points
image

### CODE REVIEW

File: next-app/styles/vars/sizes.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+$safe-col-width: 120rem;
```
Comment (PRTTMPRPHT) [inline next-app/styles/vars/sizes.scss:1]: This will equate to 1200px on a 1080p monitor. Use this size as a reference for the main column of our page design.

Import using `@import "vars/sizes";` in any `.scss` file.

File: next-app/styles/vars/mixins.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
+@mixin flex-row($center-horizontal: false, $center-vertical: false) {
```
Comment (PRTTMPRPHT) [inline next-app/styles/vars/mixins.scss:1]: These mixins will solve 95% of your layouting needs. Use them.

Import using `@import "vars/mixins";` in any `.scss` file.

Use like this: 

```
.my-class {
   @include flex-row(false, true); // Don't center on x-axis, but on y-axis!
}
```

File: next-app/src/app/globals.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    font-family: "Space Grotesk", sans-serif;
+    width: 100%;
+    min-height: 100vh;
+    padding: 0;
+    margin: 0;
+
+    // To use REM, we have to specify a base font size on the root.
+    // We make it easy for ourselves by using 10px as a reference size.
+    // On a monitor >= 1080p, 1rem equates to 10px :)
+    font-size: 10px;
```
Comment (PRTTMPRPHT) [inline next-app/src/app/globals.scss:17]: THIS is the definition of what `1rem` equates to!

REM are meant to be used in a similar way to `px`, and their actual value depends on the font size on `body`. As you can see below this, we manipulate this size now depending on the screen size

This allows for very elegant scaling of elements on smaller screens because it automatically affects all the `rem`. You just always treat `1rem` like `10px` during development, and the media queries will take care of smaller screens.

TL;DR: Don't use px anymore, divide the px value you want by 10 and use that as rem.

File: next-app/styles/vars/mixins.scss (hunk 2 above)
Comment (ferran98campos) [reply next-app/styles/vars/mixins.scss:1]: I didn't know this even existed. Great!

**Review by ferran98campos — APPROVED** [review]
Looks great to me! I would like to sit down together one day and look at the scss, so I can get a better idea of its potential so I can actually use it in future projects :)
